test/models/report_test.rb

```diff
@@ -3,7 +3,32 @@
 require 'test_helper'
 
 class ReportTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  include Devise::Test::IntegrationHelpers
+
+  setup do
+    @report = reports(:alice_report)
+    @alice = users(:alice)
+    @bob = users(:bob)
+    sign_in @alice
+  end
+
+  test '#editable?' do
+    report = Report.new(title: 'hoge', content: 'fuga', user: @alice)
+    assert report.editable?(@alice)
+    assert_not report.editable?(@bob)
+  end
+
+  test '#creat_on' do
+    assert_equal Date.current, @report.created_on
+  end
+
+  test '#言及した日報が正しくmentioning_reportsに追加されるかのテスト' do
+    not_mention = Report.new(title: 'mention_test', content: 'hoge', user: @alice)
+    not_mention.save!
+    assert_empty not_mention.mentioning_reports
+    mention = Report.new(title: 'not_mention_test', content: "http://localhost:3000/reports/#{@report.id}\n", user: @bob)
+    mention.save!
+    assert_equal 1, mention.mentioning_reports.size
+    assert_equal @report, mention.mentioning_reports[0]
```

Comment:
```suggestion
    assert_equal [@report], mention.mentioning_reports
```

配列同士を比較して完全一致するかどうかを見れば、sizeの検証は不要になります。
（ただし2件以上ある場合は並び順まで一致しないとパスしない点に注意）

Reply:
修正しました。
今回は行いませんが、2件以上であればorderで並びまで担保する必要があるのですね！